src/NzbDrone.Core/Tv/SeriesAddedService.cs

```diff
@@ -0,0 +1,140 @@
+            {
+                _logger.Debug("Ignoring Episodes with Files");
+                UnmonitorEpisodes(episodes.Where(e => e.HasFile));
+            }
+
+            if (options.IgnoreEpisodesWithoutFiles)
+            {
+                _logger.Debug("Ignoring Episodes without Files");
+                UnmonitorEpisodes(episodes.Where(e => !e.HasFile && e.AirDateUtc.HasValue && e.AirDateUtc.Value.Before(DateTime.UtcNow)));
+            }
+
+            foreach (var season in series.Seasons)
+            {
+                if (options.IgnoreSeasons.Contains(season.SeasonNumber))
+                {
+                    season.Monitored = false;
+                }
+
+                if (series.Seasons.Select(s => s.SeasonNumber).MaxOrDefault() != season.SeasonNumber &&
+                    episodes.Where(e => e.SeasonNumber == season.SeasonNumber).All(e => !e.Monitored))
+                    
+                {
+                    season.Monitored = false;
+                }
+            }
+
+            _seriesService.UpdateSeries(series);
+            _episodeService.UpdateEpisodes(episodes);
+        }
+
+        private void SearchForMissingEpisodes(Series series, List<Episode> episodes)
+        {
+            var missing = episodes.Where(e => e.Monitored && !e.HasFile).GroupBy(e => e.SeasonNumber);
+
+            var missingEpisodeIds = new List<int>();
+
+            foreach (var group in missing)
+            {
+                var seasonNumber = group.Key;
+
+                if (group.Count() > 1)
+                {
+                    //season search
+                    _commandExecutor.PublishCommand(new SeasonSearchCommand
+                                                    {
+                                                        SeriesId = series.Id,
+                                                        SeasonNumber = seasonNumber
+                                                    });
+                }
+
+                else
+                {
+                    missingEpisodeIds.AddRange(group.Select(e => e.Id));                    
+                }
+            }
+
+            if (missingEpisodeIds.Any())
+            {
+                _commandExecutor.PublishCommand(new EpisodeSearchCommand
+                                                    {
+                                                        EpisodeIds = missingEpisodeIds
+                                                    });
+            }
+
+        }
+
+        public void Handle(SeriesAddedEvent message)
+        {
+            _addSeriesOptionsCache.Set(message.Series.Id.ToString(), message.Options);
+        }
+
+        public void Handle(SeriesScannedEvent message)
+        {
+            var options = _addSeriesOptionsCache.Find(message.Series.Id.ToString());
+
+            if (options == null)
+            {
+                _logger.Debug("[{0}] Was not recently added, skipping post-add actions", message.Series.Title);
```

Comment:
Rebased, so this is hidden, but not really sure how we want to handle this exactly, this is all events that need to get handled, not commands.

I'm not a fan of if Sonarr is restarted this logic of ignoring/searching is lost though.